cpp/core/compute/Backend.h

```diff
@@ -41,21 +41,28 @@ class ResultIterator;
 
 class Backend : public std::enable_shared_from_this<Backend> {
  public:
+  Backend() {}
+  Backend(const std::unordered_map<std::string, std::string>& confMap) : confMap_(confMap) {}
   virtual ~Backend() = default;
 
   virtual std::shared_ptr<ResultIterator> GetResultIterator(
       MemoryAllocator* allocator,
       std::vector<std::shared_ptr<ResultIterator>> inputs = {}) = 0;
 
+  bool ParsePlan(const uint8_t* data, int32_t size) {
+    return ParsePlan(data, size, -1, -1, -1);
+  }
+
   /// Parse and cache the plan.
   /// Return true if parsed successfully.
-  bool ParsePlan(const uint8_t* data, int32_t size) {
+  bool ParsePlan(const uint8_t* data, int32_t size, int32_t stageId, int32_t partitionId, int64_t taskId) {
 #ifdef GLUTEN_PRINT_DEBUG
     auto buf = std::make_shared<arrow::Buffer>(data, size);
     auto maybe_plan_json = SubstraitToJSON("Plan", *buf);
     if (maybe_plan_json.status().ok()) {
       std::cout << std::string(50, '#') << " received substrait::Plan:" << std::endl;
-      std::cout << maybe_plan_json.ValueOrDie() << std::endl;
+      std::cout << "Task stageId: " << stageId << ", partitionId: " << partitionId << ", taskId: " << taskId << "; "
+                << maybe_plan_json.ValueOrDie() << std::endl;
```

Comment:
we can print a stage plan from df.explain(true). needn't to print here.

Reply:
df.explain(true) will not show the partitionId, is there a scene that user needs to get the data of one partition, not all the data of the stage?

Reply:
And sometimes, I cannot get substrait plan by `df.explanin(true)`

Reply:
When AQE is on, df.explain(true) won't print the plan before real execution like df.collect(), because the columnar transformer has not yet been applied.